### DISCUSSION TITLE
Is it possible in principle to adapt XIBs to compile on a PowerPC Mac?

### DISCUSSION THREAD

**Discussion by barracuda156** [transmission/transmission#2419] [Q&A]
I have been trying to get Transmission v. 3 built on a PowerPC Mac, and it runs into multiple XIB errors which I have already encountered when trying to build recent versions of R GUI, and so far dunno if a feasible solution exists.

To be clear, I do not expect Transmission Developer team to spend time and efforts adapting the code for PowerPC. I only hope to get a practical advice on what to try, or to be told that it is indeed technically impossible no matter what.

As of now, I have all dependencies built. However both Transmission and Transmission-X11 fail to compile (errors are different).

**Comment by DevilDimon** [comment]
The minimum OS X version that is supported now is 10.10. 
The latest Mac with a PowerPC CPU is PowerMac G5 and the last version it supports is Mac OS X 10.5.8.

You could try to backport all the APIs missing in 10.5.8, along with Xcode project versions & XIB format versions. You'll also have to move back from clang to gcc, which is more difficult now that the project uses C++17 and I don't believe PowerPC has great C++17 support.

You could also try just running the old versions if you just want to use Transmission.

**Comment by barracuda156** [comment]
@DevilDimon clang is essentially broken for PowerPC, so I use gcc for everything in any case. However I have now gcc11 natively built and working on PowerPC. Since that is the latest version, it should support everything needed.
